sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamRelNode.java

```diff
@@ -17,15 +17,77 @@
  */
 package org.apache.beam.sdk.extensions.sql.impl.rel;
 
+import java.util.List;
 import org.apache.beam.sdk.transforms.PTransform;
 import org.apache.beam.sdk.values.PCollection;
 import org.apache.beam.sdk.values.PCollectionTuple;
 import org.apache.beam.sdk.values.Row;
 import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeField;
 
 /** A {@link RelNode} that can also give a {@link PTransform} that implements the expression. */
 public interface BeamRelNode extends RelNode {
 
+  default double estimateRowSize(RelDataType rowType) {
+    List<RelDataTypeField> fieldList = rowType.getFieldList();
+
+    return fieldList
+        .stream()
+        .mapToDouble(typeField -> estimateDataTypeSize(typeField.getType()))
+        .sum();
+  }
+
+  default double estimateDataTypeSize(RelDataType type) {
```

Comment:
This can actually go onto `RowCoder` for now I think. Coders already have a place for size estimation. And you can get a `RowCoder` from any schema. You can use `CalciteUtils` to convert to `Schema`.

Reply:
Didn't notice that. I think that's a good way and I will switch to `RowCoder.estimatedSizeBytes()`

Reply:
`RowCoder.estimatedSizeBytes()` needs `Row` type. We can only use CalciteUtils to convert to Schema. Should we support `estimatedSizeBytes` by `schema` in `RowCoder`?